pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/SegmentConversionUtils.java

```diff
@@ -61,6 +65,39 @@ public class SegmentConversionUtils {
   private SegmentConversionUtils() {
   }
 
+  /**
+   * Extract non-existent segments from the given list of segments
+   * @param tableNameWithType a table name with type
+   * @param controllerBaseURI the controller base URI
+   * @param segmentNames a list of segments to check
+   * @param authProvider a {@link AuthProvider}
+   * @return a set of non-existent segment names
+   * @throws Exception when there are exceptions checking whether the given list of segments all exist or not
+   */
+  public static Set<String> extractNonExistentSegments(String tableNameWithType, URI controllerBaseURI,
```

Comment:
Instead of this, I think that it's probably better to have the function that provides the list of tables from the table.

We can simply compute `nonExistentSegments` by the following:

```
inputSegmentsListFromTaskConfig.removeAll(SegmentConversionUtils.getSegmentsForTable(xxx))
```

I feel that the getSegmentsForTable will likely be more frequently used.

Reply:
agree, changed.